src/main/java/com/google/sps/servlets/InterviewerFeedbackServlet.java

```diff
@@ -0,0 +1,147 @@
+public class InterviewerFeedbackServlet extends HttpServlet {
+  private ScheduledInterviewDao scheduledInterviewDao;
+  private PersonDao personDao;
+  private Path emailsPath =
+      Paths.get(
+          System.getProperty("user.home") + "/InterviewMe/src/main/resources/templates/email");
+
+  @Override
+  public void init() {
+    init(new DatastoreScheduledInterviewDao(), new DatastorePersonDao());
+  }
+
+  public void init(ScheduledInterviewDao scheduledInterviewDao, PersonDao personDao) {
+    this.scheduledInterviewDao = scheduledInterviewDao;
+    this.personDao = personDao;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long scheduledInterviewId = Long.parseLong(request.getParameter("interviewId"));
+    HashMap<String, String> answers = new HashMap<String, String>();
+    for (int i = 1; i < 10; i++) {
+      String template = String.format("{{question_%s}}", i);
+      String param = String.format("question%s", i);
+      answers.put(template, request.getParameter(param));
+    }
+
+    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+    String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+    // Since Users returned from the LocalUserService (in tests) do not have userIds, here we set
+    // the userId equal to a hashcode.
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+
+    if (interviewExists(scheduledInterviewId)) {
+      answers.put("{{formatted_date}}", getDateString(scheduledInterviewId));
+      if (isInterviewee(scheduledInterviewId, userId)) {
+        try {
+          sendFeedback(getInterviewerEmail(scheduledInterviewId), answers);
+        } catch (Exception e) {
+          e.printStackTrace();
+          response.sendError(500);
+          return;
+        }
+        response.sendRedirect("/scheduled-interviews.html");
+        return;
+      } else {
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+        return;
+      }
+    } else {
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+  }
+
+  private boolean interviewExists(long scheduledInterviewId) {
+    return scheduledInterviewDao.get(scheduledInterviewId).isPresent();
+  }
+
+  private boolean isInterviewee(long scheduledInterviewId, String userId) {
+    ScheduledInterview scheduledInterview = scheduledInterviewDao.get(scheduledInterviewId).get();
+    return scheduledInterview.intervieweeId().equals(userId);
+  }
+
+  private String getInterviewerEmail(long scheduledInterviewId) {
+    ScheduledInterview scheduledInterview = scheduledInterviewDao.get(scheduledInterviewId).get();
+    return personDao
+        .get(scheduledInterview.interviewerId())
+        .map(Person::email)
+        .orElse("Email not found");
+  }
+
+  private String getDateString(long scheduledInterviewId) {
```

Comment:
This is duplicated between these two servlets.

Add this as a function in ScheduledInterview?

Reply:
Done